drivers/DeviceKey.h

```diff
@@ -0,0 +1,151 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_DEVICEKEY_H
+#define MBED_DEVICEKEY_H
+
+#include "stddef.h"
+#include "stdint.h"
+#include "platform/NonCopyable.h"
+
+#if (NVSTORE_ENABLED)
+
+namespace mbed {
+/** \addtogroup drivers */
+
+#define DEVICE_KEY_16BYTE 16
+#define DEVICE_KEY_32BYTE 32
+
+enum DeviceKeyStatus {
+    DEVICEKEY_SUCCESS                     =  0,
+    DEVICEKEY_INVALID_KEY_SIZE            = -1,
+    DEVICEKEY_INVALID_KEY_TYPE            = -2,
+    DEVICEKEY_SAVE_FAILED                 = -3,
+    DEVICEKEY_ALREADY_EXIST               = -4,
+    DEVICEKEY_NOT_FOUND                   = -5,
+    DEVICEKEY_READ_FAILED                 = -6,
+    DEVICEKEY_NVSTORE_UNPREDICTABLE_ERROR = -7,
+    DEVICEKEY_ERR_CMAC_GENERIC_FAILURE    = -8,
+    DEVICEKEY_BUFFER_TO_SMALL             = -9,
+    DEVICEKEY_NO_KEY_INJECTED             = -10,
+    DEVICEKEY_INVALID_PARAM               = -11,
+    DEVICEKEY_TRNG_ERROR                  = -12,
+};
```

Comment:
We are implementing error coding/handling on platform side and I would like to see these going under that umbrella. Error coding/handling is planned for 5.9.

Reply:
What do we do if we want this feature in before 5.9?